rest/snapshot.go

```diff
@@ -69,6 +70,14 @@ func (ac *SnapshotClient) GetOptionContractSnapshot(ctx context.Context, params
 	return res, err
 }
 
+// GetOptionsChainSnapshot gets the snapshot of all options contracts for an underlying ticker. For more details see
+// https://polygon.io/docs/options/get_v3_snapshot_options__underlyingasset
+func (ac *SnapshotClient) GetOptionsChainSnapshot(ctx context.Context, params *models.GetOptionsChainParams, opts ...models.RequestOption) (*models.GetOptionsChainSnapshotResponse, error) {
```

Comment:
AFAIK this method should actually be `ListOptionsChainSnapshot` because the response is paginated and includes a `next_url`. It should return an iterator similar to our other list methods. If that's a correct assumption on my part, can we also rename the relevant types to `List*`?

Reply:
Made changes, good callout.
changed response to *iter.Iter[models.OptionsContractSnapshot] and renamed appropriately